src/main/java/org/craftercms/engine/util/logging/CircularQueueLogAppender.java

```diff
@@ -17,83 +17,83 @@
 
 package org.craftercms.engine.util.logging;
 
+import java.io.BufferedReader;
+import java.io.PrintWriter;
+import java.io.Serializable;
+import java.io.StringReader;
+import java.io.StringWriter;
 import java.text.SimpleDateFormat;
 import java.util.ArrayList;
 import java.util.Date;
 import java.util.HashMap;
 import java.util.Iterator;
 import java.util.List;
 import java.util.Map;
+import java.util.Objects;
 
 import org.apache.commons.collections.Buffer;
 import org.apache.commons.collections.BufferUtils;
 import org.apache.commons.collections.buffer.CircularFifoBuffer;
 import org.apache.commons.lang3.StringUtils;
-import org.apache.log4j.AppenderSkeleton;
-import org.apache.log4j.Layout;
-import org.apache.log4j.spi.LoggingEvent;
+import org.apache.logging.log4j.core.Filter;
+import org.apache.logging.log4j.core.Layout;
+import org.apache.logging.log4j.core.LogEvent;
+import org.apache.logging.log4j.core.appender.AbstractAppender;
+import org.apache.logging.log4j.core.config.Property;
+import org.apache.logging.log4j.core.config.plugins.Plugin;
+import org.apache.logging.log4j.core.config.plugins.PluginAttribute;
+import org.apache.logging.log4j.core.config.plugins.PluginElement;
+import org.apache.logging.log4j.core.config.plugins.PluginFactory;
+import org.apache.logging.log4j.core.layout.PatternLayout;
 import org.craftercms.engine.service.context.SiteContext;
 
+import static org.apache.logging.log4j.core.Appender.ELEMENT_TYPE;
+import static org.apache.logging.log4j.core.Core.CATEGORY_NAME;
+
 /**
  *
  */
-public class CircularQueueLogAppender extends AppenderSkeleton {
+@Plugin(name = CircularQueueLogAppender.PLUGIN_NAME, category = CATEGORY_NAME, elementType = ELEMENT_TYPE)
```

Comment:
This might not be necessary anymore if Log4j2 has some type of on demand JSON appender?

Reply:
They only provide appenders that send events to external sources like JDBC, SMTP,... but not any that holds data in memory and allows to query it